### PR TITLE
FIX gracefully handle logging exceptions in Argus and Colossus

### PR DESCRIPTION
**Pull request by zeeshanakram3**
# Context

As reported by @kdembler the Distributor node crashed with restarting the ES node, or if providing the wrong credentials. This happened because the code was not gracefully the exceptions

# Fix

This PR gracefully handles logging exceptions in Argus and Colossus

### CODE REVIEW

File: distributor-node/src/services/logging/LoggingService.ts [hunk 1]
```diff
@@ -70,12 +70,21 @@ export class LoggingService {
...
+    this.esTransport?.on('error', (error) => {
+      console.error('Error in logger caught:', error)
+    })
   }
 
   public static withAppConfig(config: ReadonlyConfig): LoggingService {
     const transports: winston.LoggerOptions['transports'] = []
 
     let esTransport: ElasticsearchTransport | undefined
+    console.log('config.logs?.elastic', config.logs?.elastic)
```
Comment (kdembler) [inline distributor-node/src/services/logging/LoggingService.ts]: 🙈

File: distributor-node/package.json [hunk 2]
```diff
@@ -50,7 +50,7 @@
     "url-join": "^4.0.1",
     "winston": "^3.3.3",
```
Comment (kdembler) [inline distributor-node/package.json:51]: Should we maybe bump winston as well while we're at it?

File: distributor-node/src/services/logging/LoggingService.ts (hunk 1 above)
Comment (zeeshanakram3) [reply distributor-node/src/services/logging/LoggingService.ts]: Removed

File: distributor-node/package.json (hunk 2 above)
Comment (zeeshanakram3) [reply distributor-node/package.json:51]: Ok. I tried to update the `winston` to the most recent version and it turned out that it contains breaking changes (so will have to update code at couple of places). So unless we absolutely want to update the dependency, I would say it's fine for now. WDYT?

Comment (kdembler) [reply distributor-node/package.json:51]: yes, let's ignore in that case

**Review by kdembler — APPROVED** [review]
For some weird reason I can't get it to connect to local ES when developing, but seems that was also the case for me before this PR.

That said, tested with public ES and it indeed caught the exception and didn't crash on failed write, so all good, thanks!

**Review by mnaamani — APPROVED** [review]

**Pull request merged**